core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -338,27 +340,27 @@ synchronized final void completeExceptionally(Throwable throwable, int topologyI
 
    private class SingleKeyCollector extends Collector<Object> {
       @GuardedBy("this")
-      private final Collection<Address> owners;
+      private final HashSet<Address> owners;
```

Comment:
unfortunately, it doesn't. and unfortunately it doesn't help. I can revert the changes if you want.